flight/src/conversion.rs

```diff
@@ -0,0 +1,159 @@
+use crate::types::*;
+use anyhow::Result;
+use arrow::array::{
+    ArrayRef, BinaryBuilder, ListBuilder, PrimitiveBuilder, StringBuilder, StructArray,
+    UInt8Builder,
+};
+use arrow::datatypes::{DataType, Field};
+use arrow::record_batch::RecordBatch;
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub struct ConclusionEventBuilder {
+    event_type: UInt8Builder,
+    stream_id: BinaryBuilder,
+    stream_type: UInt8Builder,
+    controller: StringBuilder,
+    data: BinaryBuilder,
+    previous: ListBuilder<BinaryBuilder>,
+}
+
+impl Default for ConclusionEventBuilder {
+    fn default() -> Self {
+        Self {
+            event_type: PrimitiveBuilder::new(),
+            stream_id: BinaryBuilder::new(),
```

Comment:
nit, `stream_id` to `stream_cid`.  A stream ID is an encoding of the stream type + its CID, since we have the stream type explicitly as a column on the data we only want the stream CID. So we should name the column accordingly.